exporters/prometheus/src/main/java/io/opentelemetry/exporter/prometheus/PrometheusHttpServer.java

```diff
@@ -70,6 +71,12 @@ public static PrometheusHttpServerBuilder builder() {
     this.memoryMode = memoryMode;
     this.prometheusRegistry = prometheusRegistry;
     prometheusRegistry.register(prometheusMetricReader);
+    // When memory mode is REUSABLE_DATA, concurrent reads lead to data corruption. To prevent this,
+    // we configure prometheus with a single thread executor such that requests are handled
+    // sequentially.
+    if (memoryMode == MemoryMode.REUSABLE_DATA) {
```

Comment:
Should this be `executor == null && memoryMode == REUSABLE`? Or otherwise the builder should prevent setting both?

Reply:
👋 

I left a comment on the `PrometheusHttpServerBuilder` javadoc stating that if you set memory mode, `setExecutor` is ignored. We could be stricter and throw if both are set.